syntax = "proto3";

package ChatsApp.common;


enum ResponseStatus {
    SUCCESS = 0;                
    FAILURE = 1;                
    INVALID_INPUT = 2;          
    ACCOUNT_EXISTS = 3;         
    ACCOUNT_NOT_FOUND = 4;      
    UNAUTHORIZED = 5;           
}


message User {
    int32 userid = 1;              
    string username = 2;        
    string displayname = 3;    
}


message Group {
    int32 groupid = 1;               
    string groupname = 2;            
}


enum MessageStatus {
    DELIVERED = 0;              
    FAILED = 1;                 
    PENDING = 2;                
    READ = 3;                   
}


message MessageUser {
    int32 userid = 1;          
    string username = 2;        
    string sentat = 3;         
}